mutable_tree.go

```diff
@@ -201,30 +201,30 @@ func (tree *MutableTree) set(key []byte, value []byte) (orphans []*Node, updated
 	return orphans, updated
 }
 
-func (tree *MutableTree) recursiveSet(node *Node, key []byte, value []byte, orphans *[]*Node) (
-	newSelf *Node, updated bool,
+func (tree *MutableTree) recursiveSet(node ComplexNode, key []byte, value []byte, orphans *[]ComplexNode) (
+	newSelf *TreeNode, updated bool,
 ) {
 	version := tree.version + 1
 
-	if node.isLeaf() {
+	if node.Leaf() {
 		tree.addUnsavedAddition(key, NewFastNode(key, value, version))
 
-		switch bytes.Compare(key, node.key) {
+		switch bytes.Compare(key, node.Key()) {
 		case -1:
-			return &Node{
-				key:       node.key,
+			return &TreeNode{
+				key:       node.Key(),
 				height:    1,
 				size:      2,
 				leftNode:  NewNode(key, value, version),
-				rightNode: node,
+				rightNode: node.(*TreeNode),
```

Comment:
Is it possible to have the `rightNode` and `leftNode` to be`ComplexNode` interface or pass a tree node to `recursiveSet`?